modules/core/src/main/scala/ch/epfl/scala/debugadapter/internal/evaluator/RuntimeEvaluationHelpers.scala

```diff
@@ -109,25 +107,28 @@ private[evaluator] class RuntimeEvaluationHelpers(frame: JdiFrame) {
     }
 
     finalCandidates
-      .toValidation(s"Cannot find a proper method $funName with args types $args on $ref")
+      .toValidation(s"Cannot find a proper method $encodedName with args types $args on $ref")
       .map { loadClassOnNeed }
   }
 
   def methodTreeByNameAndArgs(
       tree: RuntimeTree,
       funName: String,
-      args: Seq[RuntimeEvaluableTree],
-      encode: Boolean = true
+      args: Seq[RuntimeEvaluableTree]
   ): Validation[MethodTree] = tree match {
     case ReferenceTree(ref) =>
-      methodsByNameAndArgs(ref, funName, args.map(_.`type`), encode).flatMap {
+      methodsByNameAndArgs(ref, NameTransformer.encode(funName), args.map(_.`type`)).flatMap {
         case ModuleCall() =>
           Recoverable("Accessing a module from its instanciation method is not allowed at console-level")
         case mt => toStaticIfNeeded(mt, args, tree)
       }
     case _ => Recoverable(new IllegalArgumentException(s"Cannot find method $funName on $tree"))
   }
 
+  def newInstanceTreeByArgs(cls: ClassType, args: Seq[RuntimeEvaluableTree]): Validation[NewInstanceTree] =
+    methodsByNameAndArgs(cls, "<init>", args.map(_.`type`))
```

Comment:
```suggestion
    methodsByNameAndArgs(cls, "<init>", args.map(_.`type`), encode = false)
```
You must block the encoding, otherwise it will be called as `$lessinit$greater`